x/concentrated-liquidity/lp.go

```diff
@@ -27,14 +27,23 @@ func (k Keeper) createPosition(ctx sdk.Context, poolId uint64, owner sdk.AccAddr
 	if err != nil {
 		return sdk.Int{}, sdk.Int{}, sdk.Dec{}, err
 	}
-
 	// Check if the provided tick range is valid according to the pool's tick spacing and module parameters.
-	if err := validateTickRangeIsValid(pool.GetTickSpacing(), lowerTick, upperTick); err != nil {
+	if err := validateTickRangeIsValid(pool.GetTickSpacing(), pool.GetPrecisionFactorAtPriceOne(), lowerTick, upperTick); err != nil {
 		return sdk.Int{}, sdk.Int{}, sdk.Dec{}, err
 	}
 
-	// Transform the provided ticks into their corresponding square root prices.
-	sqrtPriceLowerTick, sqrtPriceUpperTick, err := math.TicksToSqrtPrice(lowerTick, upperTick)
+	// Transform the provided ticks into their corresponding prices.
+	priceLowerTick, priceUpperTick, err := math.TicksToPrice(lowerTick, upperTick, pool.GetPrecisionFactorAtPriceOne())
```

Comment:
Should we add back a `math.TicksToSqrtPrice` utility function?

Reply:
added back here https://github.com/osmosis-labs/osmosis/pull/3874/commits/50410c42db11758c515a3c8423f0396f82da66da

Reply:
Had to revert because tests were failing and I am signing off for the not. Didnt want to block review / merge for it